results.go

```diff
@@ -100,12 +79,60 @@ func (r Result) ExtractIntoStructPtr(to interface{}, label string) error {
 	if k := t.Kind(); k != reflect.Ptr {
 		return fmt.Errorf("Expected pointer, got %v", k)
 	}
+
+	tov := reflect.ValueOf(to)
+
+	var rawitem map[string]interface{}
+	switch bodyt := r.Body.(type) {
+	case map[string]interface{}:
+		if label == "" {
+			rawitem = bodyt
+		} else {
+			rawmap := bodyt[label]
+			switch rawmapt := rawmap.(type) {
+			case map[string]interface{}:
+				rawitem = rawmapt
+			case nil:
+				return nil
+			default:
+				return fmt.Errorf("unsupported type: %T", rawmapt)
+			}
+		}
+	default:
+		return fmt.Errorf("unsupported type for r.Body: %T", r.Body)
+	}
+
+	b, err := json.Marshal(rawitem)
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(b, to)
+	if err != nil {
+		return err
+	}
+
```

Comment:
I'll add some comments and unit tests. Basically, the first `json.Unmarshal` will handle any non response objects the don't have embedded structs. The other is specifically for embedded structs.

When a struct is embedded, the parent will inherit a method defined on the embedded struct unless that same method is defined on the parent.